### PR TITLE
Added new Class and new Interface

### PR DESCRIPTION
**Pull request by Cosmin-Crisan**
Added new class called constants and new  interface called ConstantsInterface

### CODE REVIEW

File: out/production/Typing-Practice/.gitignore [hunk 1]
```diff
@@ -0,0 +1,3 @@
+/bussinessLogic/
```
Comment (danalexandrunicoara) [inline out/production/Typing-Practice/.gitignore]: Why did you add this file?

File: src/bussinessLogic/Constants.java [hunk 2]
```diff
@@ -0,0 +1,18 @@
+package bussinessLogic;
+
+import interfaces.ConstantsInterface;
+
+public class Constants implements ConstantsInterface {
```
Comment (danalexandrunicoara) [inline src/bussinessLogic/Constants.java]: The constant class could be created without the interface, in this case.
All of her properties can be set as public constants.
In this case the class could look like the following.

```
public class Constants {
    /**
     * the number of letters in the alphabet
     */
    public static final int NUMBER_OF_LETTERS = 3;

    /** 
      * int for calculating how many times a letter is displayed in the evaluation sequence
      */
    public static final int ALPHABETMULTIPLIER = 1;
}
```

This way, there are fewer lines of code for those constants.

_Note_: In Java constants are declared all caps.

File: src/interfaces/MapManager.java [hunk 3]
```diff
@@ -37,7 +37,7 @@
     /**
      * calculates the average time for each char
      */
-    public void calculateAverage();
+    void calculateAverage();
```
Comment (danalexandrunicoara) [inline src/interfaces/MapManager.java:40]: 👍

File: src/main/StartUp.java [hunk 4]
```diff
@@ -70,9 +68,9 @@ public void windowClosing(WindowEvent windowEvent) {
 
 
     // calculates the reaction time
-    private static void calculateElapsedTime(Instant start, Instant end) {
+    private static int calculateElapsedTime(Instant start, Instant end) {
```
Comment (danalexandrunicoara) [inline src/main/StartUp.java:69]: 👍

File: src/main/StartUp.java [hunk 5]
```diff
@@ -70,9 +68,9 @@ public void windowClosing(WindowEvent windowEvent) {
...
         Duration timeElapsed = Duration.between(start, end);
-        elapsedTime = (int) timeElapsed.toMillis();
+        return (int) timeElapsed.toMillis();
```
Comment (danalexandrunicoara) [inline src/main/StartUp.java:71]: 👍

File: src/main/StartUp.java [hunk 6]
```diff
@@ -83,13 +81,12 @@ private static void runProgram(char[] array) {
         textField.addKeyListener(new KeyAdapter() {
             public void keyPressed(KeyEvent e) {
 
-                typedChar = e.getKeyChar();
+                char typedChar = e.getKeyChar();
```
Comment (danalexandrunicoara) [inline src/main/StartUp.java:82]: 👍

File: out/production/Typing-Practice/.gitignore (hunk 1 above)
Comment (Cosmin-Crisan) [reply out/production/Typing-Practice/.gitignore]: I guess it's a result of my struggle to synchronize eclipse with git to work just as my intelliJ IDE works now. :smile: I must have missed it when I made the commit

File: src/bussinessLogic/Constants.java (hunk 2 above)
Comment (Cosmin-Crisan) [reply src/bussinessLogic/Constants.java]: :+1:

Comment (Cosmin-Crisan) [reply src/bussinessLogic/Constants.java]: One question: should I move the RANDOM constant from `ManageArray` to the `Constants` class or leave it there? (RANDOM is used only locally in  `ManageArray` class)

**Comment by danalexandrunicoara** [comment]
It can stay in ManageArray class, for now.   From: Cosmin CrisanSent: Tuesday, 6 April 2021 12:50To: Cosmin-Crisan/Typing-PracticeCc: Nicoara Dan-Alexandru; Review requestedSubject: Re: [Cosmin-Crisan/Typing-Practice] Added new Class and new Interface (#8) @Cosmin-Crisan commented on this pull request.In src/bussinessLogic/Constants.java:> @@ -0,0 +1,18 @@+package bussinessLogic;++import interfaces.ConstantsInterface;++public class Constants implements ConstantsInterface {One question: should I move the RANDOM constant from ManageArray to the Constants class or leave it there? (RANDOM is used only locally in ManageArray class)—You are receiving this because your review was requested.Reply to this email directly, view it on GitHub, or unsubscribe.

File: src/bussinessLogic/Constants.java (hunk 2 above)
Comment (danalexandrunicoara) [reply src/bussinessLogic/Constants.java]: It can stay in `ManageArray` class, for now.

File: src/bussinessLogic/ManageArray.java [hunk 7]
```diff
@@ -37,11 +35,10 @@
      *
      * @param manageMap
      */
-    public ManageArray(MapManager manageMap, ConstantsInterface constant) {
+    public ManageArray(MapManager manageMap) {
         this.manageMap = manageMap;
-        this.constant = constant;
-        this. numberOfLetters = constant.getNumberOfLetters();
-        this.alphabetMultiplier = constant.getAlphabetMultiplier();
+        this. numberOfLetters = Constants.NUMBER_OF_LETTERS;
```
Comment (danalexandrunicoara) [inline src/bussinessLogic/ManageArray.java]: `Constants.NUMBER_OF_LETTERS` can be used everywhere where there is `this.numberOfLetters`.

This is valid for `this.alphabetMultiplier` as well.

File: src/bussinessLogic/Constants.java [hunk 8]
```diff
@@ -1,18 +1,13 @@
 package bussinessLogic;
 
-import interfaces.ConstantsInterface;
+public class Constants {
```
Comment (danalexandrunicoara) [inline src/bussinessLogic/Constants.java:3]: 👍

File: src/bussinessLogic/ManageArray.java (hunk 7 above)
Comment (Cosmin-Crisan) [reply src/bussinessLogic/ManageArray.java]: 👍

File: src/bussinessLogic/Constants.java (hunk 2 above)
Comment (Cosmin-Crisan) [reply src/bussinessLogic/Constants.java]: OK. Thanks!

**Review by danalexandrunicoara — APPROVED** [review]
👍

**Review by danalexandrunicoara — APPROVED** [review]
👍
